fs/span-manager/span_manager.go

```diff
@@ -162,6 +162,54 @@ func (m *SpanManager) buildAllSpans() {
 	}
 }
 
+// FetchSingleSpan invokes the reader to fetch the span in the background and cache it.
+// It is invoked by the BackgroundFetcher.
+func (m *SpanManager) FetchSingleSpan(spanID soci.SpanID) error {
```

Comment:
Wouldn't it block until the span is fetched?

Reply:
Yes, but it's called in a separate goroutine. The bg-fetcher won't be blocked on this call.